src/VisualStudio/Core/Def/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -52,28 +53,34 @@ internal sealed class FileChangeWatcher : IFileChangeWatcher
                 CancellationToken.None);
         }
 
-        private async ValueTask ProcessBatchAsync(ImmutableSegmentedList<WatcherOperation> workItems, CancellationToken cancellationToken)
+        private async ValueTask ProcessBatchAsync(ImmutableSegmentedList<WatcherOperation> operations, CancellationToken cancellationToken)
```

Comment:
>ProcessBatchAsync [](http://example.com/codeflow?start=32&length=17)

Detection of potential merges was modified a bit in this PR. 

I think the new approach is a bit simpler and definitely more efficient (TryCombineWith was potentially allocating ImmutableArrays each time it was called, which is bad when there are lots of items being merged)

Reply:
BTW I found this nice and easy to follow, so nice!